pages/components/cards.js

```diff
@@ -0,0 +1,201 @@
+    Photo: "/assets/images/2.jpg",
+    comments: [],
+  },
+  {
+    id: 3,
+    content: "Dies ist der Inhalt des dritten Beitrags.",
+    Photo: "/assets/images/3.jpg",
+    comments: [],
+  },
+];
+
+export default function Cards() {
+  const [posts, setPosts] = useState(dummyPosts);
+  const [isCommentVisible, setIsCommentVisible] = useState(false);
+
+  // to update the comment
+  const handleCommentUpdate = (event, postId) => {
+    const updatedPosts = posts.map((post) => {
+      if (post.id === postId) {
+        return {
+          ...post,
+          comment: event.target.value,
+        };
+      }
+      return post;
+    });
+    setPosts(updatedPosts);
+  };
+
+  // comment Icon visible
+  const handleCommentClick = (postId) => {
+    const updatedPosts = posts.map((post) => {
+      if (post.id === postId) {
+        return {
+          ...post,
+          isCommentVisible: !post.isCommentVisible,
+        };
+      }
+      return post;
+    });
+    setPosts(updatedPosts);
+  };
+
+  // comment function
+  const handleCommentSubmit = (event, postId) => {
+    event.preventDefault();
+    const updatedPosts = posts.map((post) => {
+      if (post.id === postId) {
+        const updatedComments = [...post.comments, post.comment];
+        return { ...post, comments: updatedComments, comment: "" };
+      }
+      return post;
+    });
+    setPosts(updatedPosts);
+  };
+
+  // likestatus Icon toggle
+  const handleLikeClick = (postId) => {
+    const updatedPosts = posts.map((post) => {
+      if (post.id === postId) {
+        return {
+          ...post,
+          likeStatus: !post.likeStatus,
+        };
+      }
+      return post;
+    });
+    setPosts(updatedPosts);
+  };
+
+  // create Click on Image
+  const doubleClickHandler = (postId) => {
+    const updatedPosts = posts.map((post) => {
+      if (post.id === postId) {
+        return { ...post, likePhotos: true, likeStatus: true };
+      }
+      return post;
+    });
+    setPosts(updatedPosts);
+
+    setTimeout(() => {
+      const updatedPosts = posts.map((post) => {
+        if (post.id === postId) {
+          return { ...post, likePhotos: false, likeStatus: true };
+        }
+        return post;
+      });
+      setPosts(updatedPosts);
+    }, 1000);
+  };
+
+  return (
+    <Card>
+      {posts.map((post) => (
+        <div key={post.id}>
+          <PhotoContainer onClick={() => doubleClickHandler(post.id)}>
+            <Photo src={post.Photo} width={100} height={100} alt={post.title} />
+            {post.likePhotos && (
+              <LikeIconImage src="/assets/images/icons/like_fill.png" />
+            )}
+          </PhotoContainer>
+
+          <p>{post.content}</p>
+          <div>
+            <IconContainer>
+              <PhotoIcon
+                src="/assets/images/icons/comment.png"
+                width={30}
+                height={30}
+                alt="comment"
+                onClick={() => handleCommentClick(post.id)}
```

Comment:
Please use a `button` to handle the click event. You can nest the `PhotoIcon` inside the button element.